src/main/java/hu/wurfel/new_school_reference/subject/SubjectService.java

```diff
@@ -1,20 +1,31 @@
 package hu.wurfel.new_school_reference.subject;
 
 import hu.wurfel.new_school_reference.base.CrudService;
+import org.modelmapper.ModelMapper;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
 
 @Service
-public class SubjectService extends CrudService<Subject,SubjectRepository> {
+public class SubjectService extends CrudService<Subject,SubjectRepository, SubjectDto> {
 
 	@Autowired
-	public SubjectService(SubjectRepository repository) {
-		super(repository);
+	public SubjectService(SubjectRepository repository, ModelMapper modelMapper) {
+		super(repository, modelMapper);
+	}
+
+	@Override
+	public SubjectDto toDto(Subject auditable) {
+		return this.mapper.map(auditable,SubjectDto.class);
+	}
+
+	@Override
+	public Subject toEntity(SubjectDto dto) {
+		return this.mapper.map(dto,Subject.class);
 	}
 
 	public List<Subject> findAllByName(String name){
-		return repo.findAllByName(name);
+		return repo.findAllByNameAndDeletedIsFalse(name);
```

Comment:
Ez amúgy egy picit lehet zavaró. 
A service method neve nem ua, mint a repo method-é. Most nem arra gondolok, hogy karakterre pontosan, hanem ha ránézel a service method nevére, akkor ok, mindent lekér név alapján, de ez nem igaz. Csak a nem törölteket kéri le. Alapvetően nem tekintheted a rendszer úgy, hogy csak public funkcionalitás lesz, ezért pl ha az igazgató jelentkezik be, akkor neki is csak a nem töröltek lesznek leküldve, ami adminisztrátori szempontból baj.